### PR TITLE
Preparing for Chef-18 and Chef-19 release branches

### PR DESCRIPTION
**Pull request by sean-simmons-progress**
This is to create the expeditor agent chef-18, as well as create the new base v19 version.
Also creates the release branch 19 (main), and chef-18. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] If `Gemfile.lock` has changed, I have used `--conservative` to do it and included the full output in the Description above.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by sean-simmons-progress** [comment]
Lets do a screen share / review on this before we merge.

**Comment by sean-simmons-progress** [comment]
Following https://github.com/chef/chef/blob/main/docs/dev/how_to/bumping_the_major_version.md#create-a-branch-to-fixup-your-new-stable-branch-for-release - will need to complete before merge.

**Review by tpowell-progress — APPROVED** [review]

**Review by neha-p6 — CHANGES_REQUESTED** [review]
We might first want to merge ohai bump changes https://github.com/chef/ohai/pull/1825/files so that ohai 19 is available on `main` there. 
Then update `VERSION` here to 19.0.0. 
Because here we are missing `Gemfile.lock`, `chef.gemspec`  and `knife.gemspec` which will point to ohai latest (i.e 19) and then run `rake dependencies:update` so that both `Gemfile.lock` and `omnibus/Gemfile.lock` are updated correctly as well.

**Review by tpowell-progress — CHANGES_REQUESTED** [review]
@sean-simmons-progress I think Neha's mentions of `main` are correct... double-check them at least.

File: omnibus/Gemfile.lock [hunk 1]
```diff
@@ -66,24 +68,24 @@ GEM
       retryable (>= 2.0, < 4.0)
       solve (~> 4.0)
       thor (>= 0.20)
-    bigdecimal (3.1.6)
-    builder (3.2.4)
-    chef (18.1.29)
+    bigdecimal (3.1.8)
+    builder (3.3.0)
+    chef (18.4.12)
```
Comment (tpowell-progress) [inline omnibus/Gemfile.lock:73]: `bundle update --conservative chef` in `omnibus` directory?

**Review by tpowell-progress — APPROVED** [review]

**Pull request merged**